### PR TITLE
Microcare MCCommsBridge and MCUBMU

### PR DESCRIPTION
**Pull request by KyleMicrocare**
MCCommsBridge and MCUBMU modules have reached a high level of development and (I believe) are ready to be mreged into the develop branch

### CODE REVIEW

**Comment by KyleMicrocare** [comment]
It seems that .classpath files have been modified again. I'll update accordingly

**Comment by KyleMicrocare** [comment]
Files now only contain significant changes. Ready for merge.

File: io.openems.edge.battery.microcare.ubmu/src/io/openems/edge/battery/microcare/ubmu/MCUBMU.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+		}
+	}
+	
+	/**
+	 * Convenenience method to construct {@link QueryTask}s for this component
+	 * @param queryCommand the command value to use when querying the MCUBMU
+	 * @param responseCommand the expected command value of the reply packet
+	 * @param responsePacketElements the elements used to create a packet structure for the reply packet to map to
+	 * @return a new {@link QueryTask}
+	 * @throws OpenemsException if the {@link QueryTask could not be constructed}
```
Comment (KyleMicrocare) [inline io.openems.edge.battery.microcare.ubmu/src/io/openems/edge/battery/microcare/ubmu/MCUBMU.java]: whoops, I'll fix this later

**Comment by sfeilmeier** [comment]
@pooran-c: I hardly have time to review this currently. Could you do a first review?

**Comment by pooran-c** [comment]
Hi @KyleMicrocare , There were some warnings, when I imported the project. If you want, i can commit those changes.

**Comment by KyleMicrocare** [comment]
> Hi @KyleMicrocare , There were some warnings, when I imported the project. If you want, i can commit those changes.

@pooran-c that would be appreciated, thanks

**Comment by pooran-c** [comment]
Hi @KyleMicrocare , could you please add the license in bnd files.

**Comment by sfeilmeier** [comment]
Hi @KyleMicrocare,

is there any progress on this Pull-Request?

**Comment by KyleMicrocare** [comment]
Hi Stefan, 

I'm just waiting on a legal opinion for licensing - my hands are currently tied. My apologies for the delay, I'll try get the cogs turning.

**Comment by KyleMicrocare** [comment]
I can't seem to be able to push to the repo - do I still have access?

**Comment by sfeilmeier** [comment]
Happy new year, Kyle!

Strange that you didn't have access. I tried some settings. Can you try again, please?

**Comment by KyleMicrocare** [comment]
I think it may just be a bad login token in SourceTree, 

> Happy new year, Kyle!
> 
> Strange that you didn't have access. I tried some settings. Can you try again, please?

Thanks Stefan, happy happy

I think it may just be my SourceTree client - I'll reset my GitHub auth token tomorrow, it's quitting time as of now.

**Comment by KyleMicrocare** [comment]
This PR is in a mergeable state - please review

**Comment by KyleMicrocare** [comment]
> This PR is in a mergeable state - please review

@pooran-c

**Comment by sfeilmeier** [comment]
Closing because of inactivity. @KyleMicrocare get in touch if you are still working on this or using it. Thanks!

**Pull request closed without merging**